lib/queue/worker.py

```diff
@@ -95,10 +99,14 @@ def execute_with_timeout(func, args, timeout_seconds: int) -> List[schemas.Messa
         Returns:
         - List[schemas.Message]: The result of the function if it completes within the timeout, otherwise an empty list.
         """
+        start_time = time.time()
         try:
             with ThreadPoolExecutor(max_workers=1) as executor:
                 future = executor.submit(func, args)
-                return future.result(timeout=timeout_seconds), True
+                result = future.result(timeout=timeout_seconds)
+                execution_time = time.time() - start_time
+                QueueWorker.log_execution_time(func.__name__, execution_time)
+                return result, True
         except TimeoutError:
             error_message = "Model respond timeout exceeded."
```

Comment:
I'd think we'd want a metric for the number of timeouts?

Reply:
will add!